addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -19,62 +19,92 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
+    },
     /**
      * @private
      * @param {Object} result
      */
     _handleCarrierUpdateResult: function (result) {
+        this._handleCarrierUpdateResultBadge(result);
         var $payButton = $('#o_payment_form_pay');
         var $amountDelivery = $('#order_delivery span.oe_currency_value');
         var $amountUntaxed = $('#order_total_untaxed span.oe_currency_value');
         var $amountTax = $('#order_total_taxes span.oe_currency_value');
         var $amountTotal = $('#order_total span.oe_currency_value');
-        var $carrierBadge = $('#delivery_carrier input[name="delivery_type"][value=' + result.carrier_id + '] ~ .badge.d-none');
-        var $computeBadge = $('#delivery_carrier input[name="delivery_type"][value=' + result.carrier_id + '] ~ .o_delivery_compute');
-        var $discount = $('#order_discounted');
-
-        if ($discount && result.new_amount_order_discounted) {
-            // Cross module without bridge
-            // Update discount of the order
-            $discount.find('.oe_currency_value').text(result.new_amount_order_discounted);
-
-            // We are in freeshipping, so every carrier is Free
-            $('#delivery_carrier .badge').text(_t('Free'));
-        }
 
         if (result.status === true) {
             $amountDelivery.text(result.new_amount_delivery);
             $amountUntaxed.text(result.new_amount_untaxed);
             $amountTax.text(result.new_amount_tax);
             $amountTotal.text(result.new_amount_total);
-            $carrierBadge.children('span').text(result.new_amount_delivery);
-            $carrierBadge.removeClass('d-none');
-            $computeBadge.addClass('d-none');
             $payButton.data('disabled_reasons').carrier_selection = false;
             $payButton.prop('disabled', _.contains($payButton.data('disabled_reasons'), true));
         } else {
-            console.error(result.error_message);
-            $computeBadge.text(result.error_message);
             $amountDelivery.text(result.new_amount_delivery);
             $amountUntaxed.text(result.new_amount_untaxed);
             $amountTax.text(result.new_amount_tax);
             $amountTotal.text(result.new_amount_total);
         }
     },
+    /**
+     * @private
+     * @param {Object} result
+     */
+    _handleCarrierUpdateResultBadge: function (result) {
+        var $carrierBadge = $('#delivery_carrier input[name="delivery_type"][value=' + result.carrier_id + '] ~ .badge.d-none');
+        var $computeBadge = $('#delivery_carrier input[name="delivery_type"][value=' + result.carrier_id + '] ~ .o_delivery_compute');
+
+        if (result.status === true) {
+             // if free delivery (`free_over` field), show 'Free', not '$0'
+             if (parseInt(result.new_amount_delivery)) {
+                 $carrierBadge.children('span').text(result.new_amount_delivery);
```

Comment:
use jsclass instead

Reply:
eg. o_wsale_delivery_badge_price?

Reply:
We can't add a class to that span, that is the span auto populated by the monetary widget.
eg: 
```xml
<span class="myclass" t-field="delivery.fixed_price" t-options='{"widget": "monetary", "from_currency": delivery.product_id.company_id.currency_id, "display_currency": website_sale_order.currency_id}'/>
```
becomes something like
```xml
<span class="myclass">
     <span>$</span>
     <span>24</span>
</span>
```